### ISSUE TITLE
NewTics (new TicTimer) requests

### ISSUE THREAD

**Issue by KBMD** [jonkb/TicTrainer-node#18]
In approximately decreasing priority order:

1. Deliver "tokens" (e.g. picture of coin, or Unicode happy face) to user on the NT DRZ schedule. Also optional (per user) chime for each token. Allow for possibility of multiple tokens per reward, separated by approximately 200 ms (for TicTrainer use, not NewTics use); this option requires space on the screen to display at least 5-20 tokens at a time.
3. Allow admin to set a new "TicTimer" mode for each *user-trainer pair*.
2. Save text files recording tic timing in a way that we can use them for NewTics analysis. Can be on server but preferably on "trainer" 
4. During "TicTimer" mode, ask trainer before each session to indicate "baseline," "verbal," "DRZ", or "NCR" mode, with defaults to 1 reward per 10 seconds tic-free and never increasing the reward rate. If NCR, also ask for either a prior DRZ session record (a la TicTimer) or a tic rate (per minute) (a la TicTrainer). Also ask for a session number.
5. Use the camera on the "user" device to send video to the "trainer" device. This must be completely separate from the TicTrainer-node programming, for privacy reasons--the TicTrainer-node project saves no PHI except for year of birth and a random date within a month or two of the actual birthdate.

**Comment by jonkb** [comment]
1-3 (& part of 4) Done, though some features should still be added. One problem is that there is no notification for the rater when the 5 minutes are up. Also that should be a setting when starting the session: how long to make it. NCR mode is not implemented yet. Also it should probably warn a rater before trying to start a session with a user without NTID.

**Comment by KBMD** [comment]
## TicTimer functionality requests, updated
* OK to ignore item 5 above.
* "no notification for the rater when the 5 minutes are up"—this is working adequately without a special notification—can ignore this.
* "a setting when starting the session: how long to make it"—agree this would be nice if it defaults to 5 min if not changed as we are always using 5 min now; one previous study used 7'30" (7.5 min) sessions, so an option for non-integer times would be useful (i.e. non-integer time in _minutes_—time in seconds can be integer).
* "NCR mode is not implemented yet"—we'll want this eventually. Can separate into its own issue if that helps.
* "Also it should probably warn a rater before trying to start a session with a user without NTID."—That would help, **but** it can't pop up this warning when the trainer clicks on e.g. "verbal," because that click is done when the session start time is indicated so it would be a slight hassle. Better would be when the admin  establishes a connection with the user before starting a new session. Also, we or others may use TicTrainer for other projects, or not use "NT" for research IDs, so the error message should be a little more generic.
